Runtime/Scripts/PersistentConnection.cs

```diff
@@ -19,50 +38,122 @@ public class PersistentConnection : MonoBehaviour
     bool hasTCPPorts;
     int latencyUpdatesCounter;
     int locationUpdatesCounter;
+    Dictionary<ServerEventType, FindCloudletEventTrigger> event_trigger_dict;
+    #region MonoBehaviour Callbacks
+
+    private void Awake()
+    {
+      event_trigger_dict = new Dictionary<ServerEventType, FindCloudletEventTrigger>();
+      event_trigger_dict.Add(ServerEventType.EventAppinstHealth, FindCloudletEventTrigger.AppInstHealthChanged);
+      event_trigger_dict.Add(ServerEventType.EventCloudletState, FindCloudletEventTrigger.CloudletStateChanged);
+      event_trigger_dict.Add(ServerEventType.EventCloudletUpdate, FindCloudletEventTrigger.CloserCloudlet);
+      event_trigger_dict.Add(ServerEventType.EventCloudletMaintenance, FindCloudletEventTrigger.CloudletMaintenanceStateChanged);
+      event_trigger_dict.Add(ServerEventType.EventLatencyProcessed, FindCloudletEventTrigger.LatencyTooHigh);
+      event_trigger_dict.Add(ServerEventType.EventError, FindCloudletEventTrigger.Error);
+      //EVENT_INIT_CONNECTION and EVENT_UNKNOWN will be logged
+      //EVENT_LATENCY_REQUEST will be processed
+    }
 
     private void OnEnable()
     {
       startStreamingEvents += StartEdgeEvents;
       config = MobiledgeXIntegration.settings.edgeEventsConfig;
     }
 
-    void StartEdgeEvents(MobiledgeXIntegration mxi)
+    private void OnApplicationPause(bool pause)
     {
-      integration = mxi;
-      EdgeEventsConfig config = MobiledgeXIntegration.settings.edgeEventsConfig;
-      
-      if (config.newFindCloudletEventTriggers.Count == 0)
+      if (pause)
       {
-        FindCloudletEvent findCloudletEvent = new FindCloudletEvent()
+        StopAllCoroutines();// stop edge events streaming
+      }
+      else
+      {
+        // resume edge events streaming
+        if (locationUpdatesRunning)
+        {
+          StartCoroutine(OnIntervalEdgeEventsLocation(integration.matchingEngine.EdgeEventsConnection));
+        }
+        if (latencyUpdatesRunning)
         {
-          trigger = FindCloudletEventTrigger.Error
-        };
-        integration.NewFindCloudletHandler(EdgeEventsStatus.error, findCloudletEvent);
-        Debug.LogError(EdgeEventsError.MissingFindCloudletTrigger.ToString());
+          StartCoroutine(OnIntervalEdgeEventsLatency(integration.matchingEngine.EdgeEventsConnection, integration.GetHost()));
+        }
+      }
+    }
+
+    private void OnApplicationQuit()
+    {
+      StopAllCoroutines();// stop edge events streaming
+    }
+
+    private void OnDestroy()
+    {
+      StopAllCoroutines();// stop edge events streaming
```

Comment:
Ok. This can be called more than once?

Reply:
Probably not, Since the scene or the game will be ended .
Also a good reminder that this object should be active with scenes transitions at least Additive Scene Loading.

Reply:
I am referring to the same StopAllCroutines() is called twice in the "Decommissioning" phase of the app: OnApplicationQuit() --> OnDisable() --> OnDestroy().

Reply:
Aha, Ok.
Yeah StopAllCoroutines() can be called more than once, like the following:

OnEnable() -> StartEdgeEventsCoroutines() -> OnApplicationPause(true) -> **StopAllCoroutines()** ->  OnApplicationPause(false) -> StartEdgeEventsCoroutines() -> OnDestroy() -> **StopAllCoroutines()** -> ||